### PR TITLE
make the chat example a runnable test

### LINKED ISSUE #71: Converting examples/chat/chat.py into a pytest test case
The example at `examples/chat/chat.py` is a useful benchmark of correctness. It would be nice to have this as a runnable `pytest` test case (maybe chat over a few messages in varying orders?) to avoid running it manually upon code changes.

### PR DESCRIPTION
**Pull request by zaibon**
fixes #71

### CODE REVIEW

File: tests/examples/test_chat.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import sys
+
+import click
+import multiaddr
+import pytest
+
+from libp2p.libp2p import *
+from peer.peerinfo import info_from_p2p_addr
+
+# TODO: change once muxed_connection supports extracting protocol id from messages
```
Comment (zixuanzh) [inline tests/examples/test_chat.py]: I believe this has been enabled with `protocol_muxer` which is equivalent of `multistream-select`. I will let @stuckinaboot shed more light on this.

Comment (zaibon) [reply tests/examples/test_chat.py]: haaa could be indeed this has been fixed since then. I had this code laying around for a while. I'll check. Thanks

Comment (zaibon) [reply tests/examples/test_chat.py]: So I tested and yes indeed protocol id is now passed during negotiation. So I updated the code and added a test for this too.

Comment (stuckinaboot) [reply tests/examples/test_chat.py]: @zaibon great thanks, if you have any questions about negotiation feel free to let me know.

File: tests/examples/test_chat.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            await stream.write(message)
+        await stream.close()
+
+    host_a.set_stream_handler(PROTOCOL_ID, stream_handler)
+
+    # Start a stream with the destination.
+    # Multiaddress of the destination peer is fetched from the peerstore using 'peerId'.
+    stream = await host_b.new_stream(host_a.get_id(), [PROTOCOL_ID])
+    received = []
+    # while True: Seems the close stream event from the other host is not received
```
Comment (robzajac) [inline tests/examples/test_chat.py]: Making a note that stream close functionality is not fully implemented yet - #92

**Pull request merged**